src/circuit.rs

```diff
@@ -92,6 +92,64 @@ pub struct Cell {
     column: Column<Any>,
 }
 
+/// An assigned cell.
+#[derive(Clone, Debug)]
```

Comment:
Something neat I probably already knew but [just became aware of](https://stegosaurusdormant.com/understanding-derive-clone/) is that if you add `#[derive(Clone)]` to a struct with generics, Rust will implement it in a way that the struct is `Clone` when its type parameters are all `Clone`.

I think it's okay to have `impl Clone for AssignedCell`, because the underlying `Cell` is a pointer to the actual cell, so what we'd be cloning is the pointer in addition to the value, which is fine to do without layouter awareness. I don't believe this will be confused with `AssignedCell::copy_advice`, which clones only the value and assigns a new `Cell`, because `Clone` can't be used to satisfy a constraint (as the cell will be in the wrong location).